### ISSUE TITLE
Appending items to list with_items only leaves the last item

### ISSUE THREAD

**Issue by kustodian** [ansible/ansible#39110]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
Core

##### ANSIBLE VERSION
```
ansible-playbook 2.5.1 (detached HEAD ed473d792b) last updated 2018/04/21 00:11:57 (GMT +200)
```

##### SUMMARY
When appending items to a list while using a `with_items` loop it only leaves the last item, instead of appending them all. This issue only exists in `2.5.1`, everything works in `2.5.0`, `devel` and all older versions, so the bug is introduced in `2.5.1`.

##### STEPS TO REPRODUCE
Run this playbook:
```yaml
---
- hosts: localhost
  vars:
    l1: []
  tasks:
    - set_fact:
        l1: '{{ l1 + [ item ] }}'
      with_items:
        - 1
        - 2

    - debug:
        var: l1
```

##### EXPECTED RESULTS
`l1` should return a list of items 1, 2:
```
TASK [debug] ******************************************************************************************************************************************************************************
ok: [localhost] => {
    "l1": [
        1,
        2
    ]
}

```

##### ACTUAL RESULTS
But it only returns the last item as the only item of the list:
```
TASK [debug] ******************************************************************************************************************************************************************************
ok: [localhost] => {
    "l1": [
        2
    ]
}

```

**Comment by kustodian** [comment]
This is already fixed in `devel` by this commit https://github.com/ansible/ansible/commit/2afb1090b14c88093a0a6bb0ab59e029f57eb599, so that commit just needs to be cherry-picked into `stable-2.5`.

**Comment by bcoca** [comment]
This has been resolved in https://github.com/ansible/ansible/pull/38302 and will be included in the 2.5.2 release.

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project
